python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -463,6 +466,63 @@ def zip(self, other):
                               "The two SparkXShards should have the same number of elements "
                               "in each partition")
 
+    def to_sparkdf(self):
+        def f(iter):
+            for pdf in iter:
+                np_records = pdf.to_records(index=False)
+                return [r.tolist() for r in np_records]
+
+        def getSchema(iter):
+            for pdf in iter:
+                return [pdf.columns.values]
+
+        rdd = self.rdd.mapPartitions(f)
+        column = self.rdd.mapPartitions(getSchema).first()
+        df = rdd.toDF(list(column))
+        return df
+
+    def to_pandas_df(self, df):
+        def to_pdf(columns):
+            def f(iter):
+                import pandas as pd
+                data = list(iter)
+                pd_df = pd.DataFrame(data, columns=columns)
+                return [pd_df]
+
+            return f
+
+        pd_rdd = df.rdd.mapPartitions(to_pdf(df.columns))
+        return pd_rdd
+
+
+    def label_encoder(self, inputColumn, outputColumn):
+        df = self.to_sparkdf()
+        from pyspark.ml.feature import StringIndexer
+        indexer = StringIndexer(inputCol=inputColumn, outputCol=outputColumn)
```

Comment:
Then in this case can it be merge with Friesian's feature_encode?
https://github.com/intel-analytics/BigDL/blob/main/python/friesian/src/bigdl/friesian/feature/table.py#L1000
Since StringIndexer seems to have poor performance when the dataset is large